### PR TITLE
Add http-port option

### PR DESCRIPTION
**Pull request by shayne-fletcher**
- Add `http-port` option;
- Add a simple `GET` endpoint, ~"test"~ "health".

### CODE REVIEW

File: triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala [hunk 1]
```diff
@@ -168,18 +168,24 @@ object Server {
...
-      // List triggers currently running for the given party
       get {
-        path("list") {
-          entity(as[ListParams]) { params =>
-            {
-              val triggerList =
-                triggersByParty.getOrElse(params.party, Set()).map(_.toString).toList
-              val result = JsObject(("triggerIds", triggerList.toJson))
-              complete(successResponse(result))
+        // Convenience endpoint for tests
```
Comment (cocreature) [inline triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala:172]: How about giving this a bit more sensible name. E.g. `/healthz` is a pretty common name for health check endpoints and lots of tooling defaults to it.

Comment (shayne-fletcher) [reply triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala:172]: I didn't know that :)

Comment (cocreature) [reply triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala:172]: One example https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/ but if you search for it you can find lots of other examples from prometheus and whatnot.

Comment (shayne-fletcher) [reply triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala:172]: Quick Google suggests `health`. That's what you meant right?

Comment (cocreature) [reply triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala:172]: I did mean `healthz` see the link but `health` works as well. Not sure what is more common in general.

File: triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceConfig.scala [hunk 2]
```diff
@@ -27,6 +28,11 @@ object ServiceConfig {
       .action((f, c) => c.copy(darPath = Some(Paths.get(f))))
       .text("Path to the dar file containing the trigger")
 
+    opt[Int]("http-port")
+      .required()
```
Comment (cocreature) [inline triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceConfig.scala:32]: Do we really want to require this? I think having some well-known port ala 6865 for sandbox or 7500 for the json API would be useful as a default.

Comment (shayne-fletcher) [reply triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceConfig.scala:32]: Fair. I'll go with 8080.

File: triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala [hunk 3]
```diff
@@ -168,18 +168,25 @@ object Server {
...
-            {
-              val triggerList =
-                triggersByParty.getOrElse(params.party, Set()).map(_.toString).toList
-              val result = JsObject(("triggerIds", triggerList.toJson))
-              complete(successResponse(result))
+        // Convenience endpoint for tests (roughly follow
+        // https://tools.ietf.org/id/draft-inadarei-api-health-check-01.html).
+        concat(
+          path("health") {
+            complete((StatusCodes.OK, JsObject(("status", "pass".toJson))))
```
Comment (rohanjr) [inline triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala:176]: Sorry I just saw this, but this response format is not quite right. The `"status"` field is supposed to mirror the status code, and there should be a `"result"` field with whatever successful response body you want, i.e. `{"status": 200, "result": value}`. I suggest using the `successResponse` helper function as in the `list` command, where you pass in the JSON `value` you want (could be the string "pass" in this case).

Comment (shayne-fletcher) [reply triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala:176]: > Sorry I just saw this, but this response format is not quite right. The `"status"` field is supposed to mirror the status code, and there should be a `"result"` field with whatever successful response body you want, i.e. `{"status": 200, "result": value}`. I suggest using the `successResponse` helper function as in the `list` command, where you pass in the JSON `value` you want (could be the string "pass" in this case).

This case intentionally breaks with our internal conventions in order to conform to the referenced RFC (internet standard proposal).

**Pull request merged**